src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -221,12 +221,28 @@ def save_registration_data(
                     submission, options, documents_client
                 )
 
-            if not registration_data.attachment_urls:
-                # TODO turn attachments into a dictionary when giving users more options than
-                # just urls.
-                registration_data.attachment_urls = register_submission_attachments(
-                    submission, options, documents_client
-                )
+            if options["informatieobjecttype_attachment"]:
+                existing = [
+                    o.submission_file_attachment
+                    for o in ObjectsAPISubmissionAttachment.objects.filter(
+                        submission_file_attachment__submission_step__submission=submission
+                    )
+                ]
+
+                objs: list[ObjectsAPISubmissionAttachment] = []
+
+                for attachment in submission.attachments:
+                    if attachment not in existing:
+                        objs.append(
+                            ObjectsAPISubmissionAttachment(
+                                submission_file_attachment=attachment,
+                                document_url=register_submission_attachment(
+                                    submission, attachment, options, documents_client
+                                ),
+                            )
+                        )
```

Comment:
(please create a follow up issue for this)

This can still cause (repeated) failures - if you have two documents and the first succeeds, while the second does not, the for-loop will crash out and it will never be persisted that the first document creation actually succeeded.

you'll probably want to `try:` around this to make sure that you can create the model instances for whatever's in `objs` in the `finally` block:

```py
try:
    for attachmenet in ...:
        # do the stuff
except:
    raise
finally:
    ObjectsAPISubmissionAttachment.objects.bulk_create(objs)
```

Reply:
Created #4041